components/fxa-client/android/src/main/java/mozilla/appservices/fxaclient/AccountEvent.kt

```diff
@@ -0,0 +1,39 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.appservices.fxaclient
+
+data class TabData(
+    val title: String,
+    val url: String
+)
+
+// https://proandroiddev.com/til-when-is-when-exhaustive-31d69f630a8b
+val <T> T.exhaustive: T
+    get() = this
+
+sealed class AccountEvent {
+    class TabReceived(val from: Device?, val entries: Array<TabData>) : AccountEvent()
```

Comment:
Tab**s**Received would be more accurate?

Reply:
+1 to `TabsReceived` FWIW

Reply:
Nope `TabReceived` corresponds to 1 tab (with all its history):
https://searchfox.org/mozilla-central/rev/ee3905439acbf81e9c829ece0b46d09d2fa26c5c/services/fxaccounts/FxAccountsCommands.js#156-158
I will document the parameter and rename TabData to TabEntry